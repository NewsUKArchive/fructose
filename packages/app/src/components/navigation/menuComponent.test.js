/* globals describe it expect */
import React from "react";
import { Text } from "react-native";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import MenuComponent from "./menuComponent";

configure({ adapter: new Adapter() });

describe("Menu Component", () => {
  const expectedComponentList = ["item1", "item2"];
  const expectedEvents = { emit: jest.fn() };
  const expectedChildren = <Text>Test Text</Text>;
  const componentUnderTest = (
    <MenuComponent
      componentList={expectedComponentList}
      events={expectedEvents}
    >
      {expectedChildren}
    </MenuComponent>
  );

  it("renders specified component", () => {
    const wrapper = shallow(componentUnderTest);
    expect(wrapper).toMatchSnapshot();
  });

  it("emits correct load event on menu item press", () => {
    const expectedComponentId = "component1";
    const wrapper = shallow(componentUnderTest);
    wrapper.instance().onMenuItemPress(expectedComponentId);
    expect(expectedEvents.emit).toHaveBeenCalledWith(
      "load",
      expectedComponentId
    );
  });
});
