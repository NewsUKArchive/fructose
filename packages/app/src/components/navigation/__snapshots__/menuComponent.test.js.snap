// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Component renders specified component 1`] = `
<SideMenu
  animationFunction={[Function]}
  animationStyle={[Function]}
  autoClosing={true}
  bounceBackOnOverdraw={true}
  disableGestures={false}
  edgeHitWidth={60}
  hiddenMenuOffset={0}
  isOpen={false}
  menu={
    <MenuList
      menuHeader="Components"
      menuItems={
        Array [
          "item1",
          "item2",
        ]
      }
      onMenuItemPress={[Function]}
    />
  }
  menuPosition="left"
  onChange={[Function]}
  onMove={[Function]}
  onSliding={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  openMenuOffset={500}
  style={
    Object {
      "backgroundColor": "#DCDCDC",
    }
  }
  toleranceX={10}
  toleranceY={10}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Test Text
  </Text>
</SideMenu>
`;
